x-pack/plugins/apm/server/routes/alerts/rule_types/transaction_error_rate/get_transaction_error_rate_chart_preview.ts

```diff
@@ -39,25 +38,32 @@ export async function getTransactionErrorRateChartPreview({
   config: APMConfig;
   apmEventClient: APMEventClient;
   alertParams: AlertParams;
-}): Promise<TransactionErrorRateChartPreviewResponse> {
+}): Promise<PreviewChartResponse> {
   const {
     serviceName,
     environment,
     transactionType,
     interval,
-    start,
     end,
     transactionName,
+    groupBy: groupByFields,
   } = alertParams;
 
   const searchAggregatedTransactions = await getSearchTransactionsEvents({
     config,
     apmEventClient,
     kuery: '',
-    start,
-    end,
   });
 
+  const allGroupByFields = getAllGroupByFields(
+    ApmRuleType.TransactionErrorRate,
+    groupByFields
+  );
+
+  const intervalAsSeconds = getIntervalInSeconds(interval);
+  const intervalAsMs = intervalAsSeconds * 1000;
+  const start = end - intervalAsMs;
```

Comment:
What about the UI just specifying `start`, `end` and `interval` when calling the server?

```suggestion
  const intervalAsSeconds = getIntervalInSeconds(interval);
  const intervalAsMs = intervalAsSeconds * 1000;
  const start = end - intervalAsMs;
```